### PR TITLE
[HOPSWORKS-1109] Experiments and models v2

### PR DESCRIPTION
**Pull request by robzor92**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [ ] HOPSWORKS JIRA issue has been opened for this PR
- [ ] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by robzor92** [comment]
https://github.com/logicalclocks/hops-testing/pull/581

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/models/ModelsController.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      if(modelDTO.getMetrics() != null && !modelDTO.getMetrics().getAttributes().isEmpty()) {
+        modelSummaryStr = castMetricsToDouble(modelSummaryStr);
+      }
+
+      byte[] model = modelSummaryStr.getBytes(StandardCharsets.UTF_8);
+
+      EnumSet<XAttrSetFlag> flags = EnumSet.noneOf(XAttrSetFlag.class);
+      flags.add((XAttrSetFlag.CREATE));
+
+      dfso.setXAttr(modelPath, "provenance.model_summary", model, flags);
```
Comment (o-alex) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/models/ModelsController.java]: Use your ModelsBuilder. MODEL_SUMMARY_XATTR_NAME

Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/models/ModelsController.java]: fixed

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkYarnRunnerBuilder.java [hunk 2]
```diff
@@ -127,7 +127,7 @@ YarnRunner getYarnRunner(Project project, String jobUser, DistributedFileSystemO
       project.getName().toLowerCase() + "," + job.getName() + "," + job.getId() + "," + YarnRunner.APPID_PLACEHOLDER);
   
     Map<String, String> finalJobProps = flinkConfigurationUtil
-      .getFrameworkProperties(project, job.getJobConfig(), settings, jobUser, null, null, extraJavaOptions);
+      .setFrameworkProperties(project, job.getJobConfig(), settings, jobUser, null, extraJavaOptions);
```
Comment (ErmiasG) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkYarnRunnerBuilder.java]: Is this thing correct? get to set

File: hopsworks-web/yo/app/scripts/controllers/experimentDetailCtrl.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
```
Comment (ErmiasG) [inline hopsworks-web/yo/app/scripts/controllers/experimentDetailCtrl.js]: 2019

File: hopsworks-web/yo/app/scripts/controllers/tensorBoardCtrl.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
```
Comment (ErmiasG) [inline hopsworks-web/yo/app/scripts/controllers/tensorBoardCtrl.js]: 2019

Comment (robzor92) [reply hopsworks-web/yo/app/scripts/controllers/tensorBoardCtrl.js]: fixed

File: hopsworks-web/yo/app/scripts/controllers/experimentDetailCtrl.js (hunk 3 above)
Comment (robzor92) [reply hopsworks-web/yo/app/scripts/controllers/experimentDetailCtrl.js]: fixed

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/FilterBy.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
```
[the remark this answers is not in the document]
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/FilterBy.java]: fixed

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/ExperimentsBuilder.java [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+        experimentDTO.setOptimizationKey(experimentSummary.getOptimizationKey());
+        experimentDTO.setJobName(experimentSummary.getJobName());
+        experimentDTO.setAppId(experimentSummary.getAppId());
+        experimentDTO.setBestDir(experimentSummary.getBestDir());
+        experimentDTO.setEnvironment(experimentSummary.getEnvironment());
+        experimentDTO.setProgram(experimentSummary.getProgram());
+        experimentDTO.setTensorboard(tensorBoardBuilder.build(uriInfo,
+            resourceRequest.get(ResourceRequest.Name.TENSORBOARD), project, fileProvenanceHit.getMlId()));
+        experimentDTO.setResults(experimentResultsBuilder.build(uriInfo,
+            resourceRequest.get(ResourceRequest.Name.RESULTS), project, fileProvenanceHit.getMlId()));
```
[the remark this answers is not in the document]
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/ExperimentsBuilder.java]: removed provenance expansion

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/ExperimentExpansions.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    } else {
+      name = ResourceRequest.Name.valueOf(queryParam.toUpperCase());
+    }
+    switch (name) {
+      case RESULTS:
+        resourceRequest = new ExperimentResultsResourceRequest(name, queryParam);
+        break;
+      case TENSORBOARD:
+        resourceRequest = new TensorBoardResourceRequest(name, queryParam);
+        break;
```
[the remark this answers is not in the document]
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/ExperimentExpansions.java]: removed provenance expansion

File: hopsworks-api/pom.xml [hunk 8]
```diff
@@ -193,7 +193,12 @@
       <artifactId>commons-httpclient</artifactId>
       <version>3.1</version>
     </dependency>
-
+    <dependency>
```
[the remark this answers is not in the document]
Comment (robzor92) [reply hopsworks-api/pom.xml]: Yes I use it in the ModelConverter and ExperimentConverter to create JAXBContext https://github.com/robzor92/hopsworks/blob/3ee27633fc6ca944516bae1044c233876c9cf755/hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/ExperimentConverter.java#L23

**Pull request merged**